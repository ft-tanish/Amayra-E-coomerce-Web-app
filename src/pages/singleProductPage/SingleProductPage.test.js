import React from 'react'
import SingleProductPage from './SingleProductPage';
import { act } from "react-dom/test-utils";
import { render, screen } from "@testing-library/react"
import { BrowserRouter, useParams, Link, useNavigate } from 'react-router-dom'
import { useCart } from "../../context/cart-context";
import { useWishlist } from "../../context/wishlist-context";



jest.mock('../../context/cart-context', () => ({
    useCart: () => { return { cartProduct: [], setCartProduct: jest.fn } }
}));

jest.mock('../../context/wishlist-context', () => ({
    useWishlist: () => { return { wishListProduct: [], setWishListProduct: jest.fn } }
}));
describe('SingleProductPage', () => {

    it("should render product", () => {
        const { container } = render(
            <BrowserRouter>
                <SingleProductPage />
            </BrowserRouter>
        );
        expect(container).toMatchSnapshot()
    });
    it("should render product 1", () => {
        act(() => {
            render(
                <BrowserRouter>
                    <SingleProductPage />
                </BrowserRouter>
            );
        });
        expect(screen.queryByText("Women's Collection")).toBeTruthy();

        // expect(screen.queryByText("Want Our Assistance ?")).toBeTruthy();

    });

});