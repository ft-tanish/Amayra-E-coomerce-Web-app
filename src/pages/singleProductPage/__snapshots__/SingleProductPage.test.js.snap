// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleProductPage should render product 1`] = `
<div>
  <div
    class="single-product-sec-cont"
  >
    <div
      class="single-product-cont"
    >
      <div
        class="single-product-img-sec"
      >
        <div
          class="img-sec-small"
        >
          <div
            class="smallimg-item"
          >
            <img
              alt="item"
            />
          </div>
          <div
            class="smallimg-item"
          >
            <img
              alt="item"
            />
          </div>
          <div
            class="smallimg-item"
          >
            <img
              alt="item"
            />
          </div>
          <div
            class="smallimg-item"
          >
            <img
              alt="item"
            />
          </div>
        </div>
        <div
          class="img-sec-main-img"
        >
          <img
            alt="main"
          />
        </div>
      </div>
      <div
        class="single-product-description"
      >
        <div
          class="go-back"
        >
          <a
            class="go-back"
            href="/shop"
          >
            <img
              alt="arrow"
              src="/images/fill-right-arrow.png"
            />
            BACK
          </a>
        </div>
        <div
          class="star-rating"
        >
          <button
            class="off"
            id="star-btn"
            type="button"
          >
            <span
              class="star"
            >
              ★
            </span>
          </button>
          <button
            class="off"
            id="star-btn"
            type="button"
          >
            <span
              class="star"
            >
              ★
            </span>
          </button>
          <button
            class="off"
            id="star-btn"
            type="button"
          >
            <span
              class="star"
            >
              ★
            </span>
          </button>
          <button
            class="off"
            id="star-btn"
            type="button"
          >
            <span
              class="star"
            >
              ★
            </span>
          </button>
          <button
            class="off"
            id="star-btn"
            type="button"
          >
            <span
              class="star"
            >
              ★
            </span>
          </button>
        </div>
        <div
          class="product-name"
        />
        <div
          class="product-price"
        >
          <span>
            $1000
          </span>
          $
        </div>
        <p
          class="product-des"
        />
        <span
          class="outOfStockTextShow"
        >
          Out of stock
        </span>
        <div
          class="product-card-btn-cont"
        >
          <input
            placeholder="1"
            type="text"
          />
          <span>
            Out Of Stock
          </span>
        </div>
        <div
          class="add-to-wishlist"
        >
          <img
            alt="heart"
            src="/images/like.png"
          />
          add to wishlist
        </div>
      </div>
    </div>
    <div
      class="product-tags"
    >
      <div
        class="products-cat"
      >
        <span>
          Categories:
        </span>
      </div>
      <div
        class="products-cat"
      >
        <span>
          Tags:
        </span>
      </div>
    </div>
    <div
      class="review-additionInfo-cont"
    >
      <span>
        ADDITIONAL INFORMATION
      </span>
      <span>
        reviews
      </span>
    </div>
    <div
      class="additional-info"
    >
      <div
        class="additional-info-item hr-check"
      >
        <span>
          colors
        </span>
        Green, Red
      </div>
      <div
        class="additional-info-item"
      >
        <span>
          sizes
        </span>
      </div>
    </div>
    <div
      class="instagram-sec"
    >
      <div
        class="instagram-cont"
      >
        <img
          alt="insta"
          class="instapic"
          src="/images/ig/ig1.jpg"
        />
        <img
          alt="insta"
          class="instapic"
          src="/images/ig/ig2.jpg"
        />
        <img
          alt="insta"
          class="instapic"
          src="/images/ig/ig3.jpg"
        />
        <img
          alt="insta"
          class="instapic"
          src="/images/ig/ig4.jpg"
        />
        <img
          alt="insta"
          class="instapic"
          src="/images/ig/ig5.jpg"
        />
        <img
          alt="insta"
          class="instapic"
          src="/images/ig/ig6.jpg"
        />
        <img
          alt="insta"
          class="instapic"
          src="/images/ig/ig8.jpg"
        />
        <img
          alt="insta"
          class="instapic"
          src="/images/ig/ig9.jpg"
        />
        <div
          class="insta-overlay"
        >
          <a
            class="inst-follow-cont"
            href="https://www.instagram.com/ft.tanish/"
          >
            follow our instagram
          </a>
        </div>
      </div>
    </div>
    <div
      class="footer-sec"
    >
      <div
        class="footer-sec-cont"
      >
        <div
          class="footer-data-sec"
        >
          <div
            class="data-cont"
          >
            <div
              class="data-cont-head"
            >
              amayra's shop
            </div>
            <span>
              About Us
            </span>
            <span>
              Browse Products
            </span>
            <span>
              Contact US
            </span>
            <span>
              FAQs
            </span>
          </div>
          <div
            class="data-cont"
          >
            <div
              class="data-cont-head"
            >
              collections
            </div>
            <span>
              Summer 2022
            </span>
            <span>
              Women's Collection
            </span>
            <span>
              Men's Jackets
            </span>
            <span>
              Shoes
            </span>
          </div>
          <div
            class="data-cont"
          >
            <div
              class="data-cont-head"
            >
              customer care
            </div>
            <span>
              Order Tracking
            </span>
            <span>
              Privacy Policy
            </span>
            <span>
              Payments And Reaturns
            </span>
            <span>
              FAQs
            </span>
          </div>
        </div>
        <div
          class="footer-news-letter"
        >
          <div
            class="data-cont-head"
          >
            new's letter
          </div>
          <span>
            Get instant updates about our new products and special
          </span>
          <span>
            promos!
          </span>
          <div
            class="news-letter-input-box"
          >
            <input
              placeholder="your email address"
              type="text"
            />
            <button>
              susbcribe
            </button>
          </div>
          <div
            class="payments-sec"
          >
            <img
              alt="payment"
              src="/images/payment.png"
            />
          </div>
        </div>
      </div>
      <div
        class="footer-social-links"
      >
        <div
          class="footer-links-cont"
        >
          <img
            alt="twitter"
            src="/images/social/twitter.png"
          />
          <img
            alt="facebook"
            src="/images/social/facebook.png"
          />
          <img
            alt="pintrest"
            src="/images/social/pinterest.png"
          />
        </div>
      </div>
      <div
        class="footer-copyright"
      >
        © 2022 -Amayra's Shop : Made from passion for fashion in
        <img
          alt=""
          src="https://images.ctfassets.net/0nm5vlv2ad7a/6naVcO5MsOWXuYiol4l43R/9170af8985a9af2f9218cb02806eecbc/benefits-of-react.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`should render product 1`] = `
<div>
  <div
    class="single-product-sec-cont"
  >
    <div
      class="single-product-cont"
    >
      <div
        class="single-product-img-sec"
      >
        <div
          class="img-sec-small"
        >
          <div
            class="smallimg-item"
          >
            <img
              alt="item"
            />
          </div>
          <div
            class="smallimg-item"
          >
            <img
              alt="item"
            />
          </div>
          <div
            class="smallimg-item"
          >
            <img
              alt="item"
            />
          </div>
          <div
            class="smallimg-item"
          >
            <img
              alt="item"
            />
          </div>
        </div>
        <div
          class="img-sec-main-img"
        >
          <img
            alt="main"
          />
        </div>
      </div>
      <div
        class="single-product-description"
      >
        <div
          class="go-back"
        >
          <a
            class="go-back"
            href="/shop"
          >
            <img
              alt="arrow"
              src="/images/fill-right-arrow.png"
            />
            BACK
          </a>
        </div>
        <div
          class="star-rating"
        >
          <button
            class="off"
            id="star-btn"
            type="button"
          >
            <span
              class="star"
            >
              ★
            </span>
          </button>
          <button
            class="off"
            id="star-btn"
            type="button"
          >
            <span
              class="star"
            >
              ★
            </span>
          </button>
          <button
            class="off"
            id="star-btn"
            type="button"
          >
            <span
              class="star"
            >
              ★
            </span>
          </button>
          <button
            class="off"
            id="star-btn"
            type="button"
          >
            <span
              class="star"
            >
              ★
            </span>
          </button>
          <button
            class="off"
            id="star-btn"
            type="button"
          >
            <span
              class="star"
            >
              ★
            </span>
          </button>
        </div>
        <div
          class="product-name"
        />
        <div
          class="product-price"
        >
          <span>
            $1000
          </span>
          $
        </div>
        <p
          class="product-des"
        />
        <span
          class="outOfStockTextShow"
        >
          Out of stock
        </span>
        <div
          class="product-card-btn-cont"
        >
          <input
            placeholder="1"
            type="text"
          />
          <span>
            Out Of Stock
          </span>
        </div>
        <div
          class="add-to-wishlist"
        >
          <img
            alt="heart"
            src="/images/like.png"
          />
          add to wishlist
        </div>
      </div>
    </div>
    <div
      class="product-tags"
    >
      <div
        class="products-cat"
      >
        <span>
          Categories:
        </span>
      </div>
      <div
        class="products-cat"
      >
        <span>
          Tags:
        </span>
      </div>
    </div>
    <div
      class="review-additionInfo-cont"
    >
      <span>
        ADDITIONAL INFORMATION
      </span>
      <span>
        reviews
      </span>
    </div>
    <div
      class="additional-info"
    >
      <div
        class="additional-info-item hr-check"
      >
        <span>
          colors
        </span>
        Green, Red
      </div>
      <div
        class="additional-info-item"
      >
        <span>
          sizes
        </span>
      </div>
    </div>
    <div
      class="instagram-sec"
    >
      <div
        class="instagram-cont"
      >
        <img
          alt="insta"
          class="instapic"
          src="/images/ig/ig1.jpg"
        />
        <img
          alt="insta"
          class="instapic"
          src="/images/ig/ig2.jpg"
        />
        <img
          alt="insta"
          class="instapic"
          src="/images/ig/ig3.jpg"
        />
        <img
          alt="insta"
          class="instapic"
          src="/images/ig/ig4.jpg"
        />
        <img
          alt="insta"
          class="instapic"
          src="/images/ig/ig5.jpg"
        />
        <img
          alt="insta"
          class="instapic"
          src="/images/ig/ig6.jpg"
        />
        <img
          alt="insta"
          class="instapic"
          src="/images/ig/ig8.jpg"
        />
        <img
          alt="insta"
          class="instapic"
          src="/images/ig/ig9.jpg"
        />
        <div
          class="insta-overlay"
        >
          <a
            class="inst-follow-cont"
            href="https://www.instagram.com/ft.tanish/"
          >
            follow our instagram
          </a>
        </div>
      </div>
    </div>
    <div
      class="footer-sec"
    >
      <div
        class="footer-sec-cont"
      >
        <div
          class="footer-data-sec"
        >
          <div
            class="data-cont"
          >
            <div
              class="data-cont-head"
            >
              amayra's shop
            </div>
            <span>
              About Us
            </span>
            <span>
              Browse Products
            </span>
            <span>
              Contact US
            </span>
            <span>
              FAQs
            </span>
          </div>
          <div
            class="data-cont"
          >
            <div
              class="data-cont-head"
            >
              collections
            </div>
            <span>
              Summer 2022
            </span>
            <span>
              Women's Collection
            </span>
            <span>
              Men's Jackets
            </span>
            <span>
              Shoes
            </span>
          </div>
          <div
            class="data-cont"
          >
            <div
              class="data-cont-head"
            >
              customer care
            </div>
            <span>
              Order Tracking
            </span>
            <span>
              Privacy Policy
            </span>
            <span>
              Payments And Reaturns
            </span>
            <span>
              FAQs
            </span>
          </div>
        </div>
        <div
          class="footer-news-letter"
        >
          <div
            class="data-cont-head"
          >
            new's letter
          </div>
          <span>
            Get instant updates about our new products and special
          </span>
          <span>
            promos!
          </span>
          <div
            class="news-letter-input-box"
          >
            <input
              placeholder="your email address"
              type="text"
            />
            <button>
              susbcribe
            </button>
          </div>
          <div
            class="payments-sec"
          >
            <img
              alt="payment"
              src="/images/payment.png"
            />
          </div>
        </div>
      </div>
      <div
        class="footer-social-links"
      >
        <div
          class="footer-links-cont"
        >
          <img
            alt="twitter"
            src="/images/social/twitter.png"
          />
          <img
            alt="facebook"
            src="/images/social/facebook.png"
          />
          <img
            alt="pintrest"
            src="/images/social/pinterest.png"
          />
        </div>
      </div>
      <div
        class="footer-copyright"
      >
        © 2022 -Amayra's Shop : Made from passion for fashion in
        <img
          alt=""
          src="https://images.ctfassets.net/0nm5vlv2ad7a/6naVcO5MsOWXuYiol4l43R/9170af8985a9af2f9218cb02806eecbc/benefits-of-react.png"
        />
      </div>
    </div>
  </div>
</div>
`;
